src/helpers/xml-builder.js

```diff
@@ -1430,116 +1479,339 @@ const buildTableCellProperties = (attributes) => {
   return tableCellPropertiesFragment;
 };
 
-const fixupTableCellBorder = (vNode, attributes) => {
-  if (Object.prototype.hasOwnProperty.call(vNode.properties.style, 'border')) {
-    if (vNode.properties.style.border === 'none' || vNode.properties.style.border === 0) {
-      attributes.tableCellBorder = {
-        strokes: { ...setUpDirectionalBorderStroke('nil') },
-        colors: { ...setUpDirectionalBorderColor('000000') },
-      };
-    } else {
-      // eslint-disable-next-line no-use-before-define
-      const [borderSize, borderStroke, borderColor] = cssBorderParser(
-        vNode.properties.style.border
-      );
-
-      attributes.tableCellBorder = {
-        top: borderSize,
-        left: borderSize,
-        bottom: borderSize,
-        right: borderSize,
-        colors: {
-          ...setUpDirectionalBorderColor(borderColor),
-        },
-        strokes: {
-          ...setUpDirectionalBorderStroke(borderStroke),
-        },
-      };
-    }
-  }
+const fixupTableCellBorder = (vNode, attributes, tableBorderOptions = {}, rowIndexEquivalent) => {
```

Comment:
trying to parse through this and its hard to follow without prior context, can you add comments throughout this section to clarify what is happening here?